### PR TITLE
30 data scope funded research nsf

### PR DESCRIPTION
**Pull request by ampudia19**
### Description

Introduces a pipeline for retrieving NSF award data and a `ZipDataset` class for handling ZIP archives. The pipeline fetches award data using the NSF API, while `ZipDataset` enables accessing ZIP files in S3 and fetch all filenames there. These filenames correspond to award IDs and are used to call the API.

### Key Points

- **NSF Data Collection**: Fetches and processes NSF award information.
- **ZIP Dataset Handling**: Directly reads file names from ZIP archives, supporting both local and cloud storage. Note that the custom dataset can be easily modified to handle reading files, rather than just the names.

### CODE REVIEW

File: src/alphafold_impact/pipelines/data_collection_nsf/nodes.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            are the NSF PAR IDs and the values are dictionaries
+        containing the publication information (journal, year, volume,
+            authors, title, doi).
+    """
+    publication_dict = {}
+    for paper in publication_list:
+        paper_data = paper.split("~")
+        if len(paper_data) != 17:
+            logger.info("Paper data is incomplete. Skipping paper.")
+            continue
```
Comment (Jack-0-0) [inline src/alphafold_impact/pipelines/data_collection_nsf/nodes.py:38]: What % of papers are skipped?

File: src/alphafold_impact/datasets/zip_dataset.py
Comment (Jack-0-0) [inline src/alphafold_impact/datasets/zip_dataset.py:1]: Nice!

File: src/alphafold_impact/datasets/zip_dataset.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        Args:
+            filepath: The location of the image file to load / save data.
+            credentials: Credentials required to get access to the underlying filesystem.
+                E.g. for ``GCSFileSystem`` it should look like `{"token": None}`.
+        """
+        protocol, path = get_protocol_and_path(filepath)
+        self._protocol = protocol
+        self._filepath = PurePosixPath(path)
+        self._fs = fsspec.filesystem(self._protocol, **credentials)  # ,  **fs_args)
```
Comment (Jack-0-0) [inline src/alphafold_impact/datasets/zip_dataset.py:33]: Can this comment be removed?

File: src/alphafold_impact/pipelines/data_collection_nsf/nodes.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    Create publication objects from a list of paper data.
+
+    Args:
+        publication_list (List[str]): A list of paper data in the format
+            "journal~year~volume~authors~doi~...".
+
+    Returns:
+        Dict[str, str]: A dictionary of publication objects, where the keys
+            are the NSF PAR IDs and the values are dictionaries
+        containing the publication information (journal, year, volume,
```
Comment (Jack-0-0) [inline src/alphafold_impact/pipelines/data_collection_nsf/nodes.py]: The line starting with `containing` is not indented to same amount as the other lines

File: src/alphafold_impact/pipelines/data_collection_nsf/nodes.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            return award_object
+        else:
+            logger.info("No award data received for %s", award_id)
+            return {}
+    except Exception as e:  # pylint: disable=broad-except
+        logger.exception("Failed to fetch award data for %s: %s", award_id, e)
+        return {}
+
+
+# def fetch_awards_list_from_zip(zip_file_path: str) -> List[str]:
```
Comment (Jack-0-0) [inline src/alphafold_impact/pipelines/data_collection_nsf/nodes.py]: Can remove this comment?

File: src/alphafold_impact/pipelines/data_collection_nsf/nodes.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+
+def fetch_nsf_data(config: Dict[str, str], fields: List[str], award_list: List[str]):
```
Comment (Jack-0-0) [inline src/alphafold_impact/pipelines/data_collection_nsf/nodes.py]: Missing a return typehint here

**Review by Jack-0-0 — APPROVED** [review]
This looks good. I have left some comments.
It was nice to see how to implement a new dataset 🙂

File: src/alphafold_impact/pipelines/data_collection_nsf/nodes.py (hunk 1 above)
Comment (ampudia19) [reply src/alphafold_impact/pipelines/data_collection_nsf/nodes.py:38]: Not that many, these tend to refer to workshop events or conference pieces, they don't have a DOI associated as a general rule.

**Pull request merged**